js/ui/appDisplay.js

```diff
@@ -651,6 +651,7 @@ var AllView = class AllView extends BaseAppView {
                     let item = Shell.DesktopDirInfo.new(itemId);
                     if (!item) {
                         log(`Error loading folder for ${itemId}`);
+                        this._iconGridLayout.removeIcon(itemId, false);
```

Comment:
Is it safe to modify the structure while it is being iterated?

Reply:
It is not safe, but in this case we are iterating over `items` (a copy) and removing from the icon grid.

Reply:
OK, but will something unexpected happen, like (for example) re-entrantly entering the code which reads desktop files?

Reply:
It should not, also tested locally and didn't spot any issue.